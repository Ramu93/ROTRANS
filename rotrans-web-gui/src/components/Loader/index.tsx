import React from "react";
import Spinner from "react-spinner-material";

import './styles.css';

interface Props {
  visible: boolean;
}

const Loader: React.FC<Props> = ({ visible }) => (
  <div data-testid="spinner" className="custom-loader-div">
    <Spinner radius={80} color={"#6435c9"} stroke={5} visible={visible} />
  </div>
);

export default Loader;
