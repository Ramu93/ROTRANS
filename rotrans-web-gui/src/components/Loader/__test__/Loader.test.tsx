import React from "react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import Loader from "..";

describe("Loader component", () => {
  it("Matches snapshot - visible", () => {
    const tree = renderer.create(<Loader visible />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Matches snapshot - invisible", () => {
    const tree = renderer.create(<Loader visible={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
