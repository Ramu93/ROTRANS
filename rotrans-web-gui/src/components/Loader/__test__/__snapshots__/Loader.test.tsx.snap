// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component Matches snapshot - invisible 1`] = `
<div
  className="custom-loader-div"
  data-testid="spinner"
/>
`;

exports[`Loader component Matches snapshot - visible 1`] = `
<div
  className="custom-loader-div"
  data-testid="spinner"
>
  <div
    className="react-spinner-material"
    style={
      Object {
        "borderColor": "#6435c9",
        "borderWidth": 5,
        "height": 80,
        "width": 80,
      }
    }
  />
</div>
`;
